File: prototype/Flyer.java
package prototype;

public class Flyer extends MarketingMaterial {
    public Flyer(String layout, String content, String color) {
        super(layout, content, color);
    }

    @Override
    public Flyer cloneMarketingMaterial() {
        return new Flyer(this.getLayout(), this.getContent(), this.getColor());
    }
}

File: typescript/patterns/prototype/flyer.ts
class Flyer extends MarketingMaterial {
  constructor(content: string, layout: string, color: string) {
    super(content, layout, color);
  }

  clone(): MarketingMaterial {
    return new Flyer(this.getContent(), this.getLayout(), this.getColor());
  }
}
